plugins/BEdita/Core/tests/TestCase/Database/Type/BoolTypeTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2017 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+namespace BEdita\Core\Test\TestCase\Database\Type;
+
+use BEdita\Core\Database\Type\BoolType;
+use Cake\Datasource\ConnectionManager;
+use Cake\TestSuite\TestCase;
+use InvalidArgumentException;
+
+/**
+ * {@see \BEdita\Core\Database\Type\BoolType} Test Case
+ *
+ * @coversDefaultClass \BEdita\Core\Database\Type\BoolType
+ */
+class BoolTypeTest extends TestCase
+{
+    /**
+     * Data provider for `testToDatabase`.
+     *
+     * @return array
+     */
+    public function toDatabaseProvider()
+    {
+        return [
+            [
+                1,
+                true,
+            ],
+            [
+                0,
+                false,
+            ],
+            [
+                "1",
+                true,
+            ],
+            [
+                "0",
+                false,
+            ],
+            [
+                true,
+                true,
+            ],
+            [
+                false,
+                false,
+            ],
+            [
+                null,
+                null,
+            ],
+            [
+                "true",
+                true,
+            ],
+            [
+                "false",
+                false,
+            ],
+            [
+                "gustavo",
+                null,
+            ],
+            [
+                [1, 2, 3],
+                new InvalidArgumentException('Cannot convert value to bool'),
+            ]
+        ];
+    }
+
+    /**
+     * Test `toDatabase` method
+     *
+     * @param mixed $expected Expected result
+     * @param mixed $input Input data to be marshaled.
```

Comment:
are parameters in wrong order? first `$input` then `$expected`

Reply:
Usually we put expected result first, but it's not a rule nor a "common practice", as far as I can tell: it's just how data provider documentation is written in PHPUnit official docs.

Reply:
`@param` order fixed